EFCore.NamingConventions/Internal/NameRewritingConvention.cs

```diff
@@ -225,7 +236,14 @@ private void RewriteColumnName(IConventionPropertyBuilder propertyBuilder)
             // TODO: The following is a temporary hack. We should probably just always set the relational override below,
             // but https://github.com/dotnet/efcore/pull/23834
 #pragma warning disable 618
-            propertyBuilder.HasColumnName(_namingNameRewriter.RewriteName(property.GetColumnName()));
+            var table = StoreObjectIdentifier.Create(property.DeclaringEntityType, StoreObjectType.Table);
```

Comment:
Note the TODO above about this being a temporary hack that needs to be cleaned up...  But that can be scoped out of this PR.

In the meantime, regarding your specific proposal... Note that the ColumnName annotation is removed right above, so there's no need to try to find it. The below should be sufficient as equivalent to the current hack:

```c#
var baseColumnName = StoreObjectIdentifier.Create(property.DeclaringEntityType, StoreObjectType.Table) is { } tableIdentifier
    ? property.GetDefaultColumnName(tableIdentifier)
    : property.GetDefaultColumnBaseName();

propertyBuilder.HasColumnName(_namingNameRewriter.RewriteName(baseColumnName));
```

Also, as no obsolete API is used any more, you can remove the warning suppression.

Reply:
@roji Thank you for the review!